samples/cpp/image_generation/text2image_concurrency.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+#include <iostream>
+
+#include "openvino/genai/image_generation/text2image_pipeline.hpp"
+
+
+#include "imwrite.hpp"
+#include "progress_bar.hpp"
+
+int32_t main(int32_t argc, char* argv[]) try {
+    OPENVINO_ASSERT(argc >= 3, "Usage: ", argv[0], " <MODEL_DIR> '<PROMPT>' '<PROMPT>' ...");
+
+    const std::string models_path = argv[1];
+    const std::string device = "CPU";  // GPU can be used as well
+
+    std::vector<std::string> prompts;
+    for (int i = 2; i < argc; ++i) {
+        prompts.push_back(argv[i]);
+    }
+
+    ov::genai::Text2ImagePipeline pipe(models_path, device);
```

Comment:
Here in constructor we created one extra unused infer request

Reply:
@likholat  is it a big problem? if we dont create infer request, we need to hold ov::CompiledModel and code gets more complicated. I really like the idea of generating new infer request out of previous infer request (which holds ov::CompiledModel reference). What do you think?

Reply:
Let's just infer this original pipeline as well

Reply:
Please check https://github.com/openvinotoolkit/openvino.genai/pull/2190#discussion_r2102144487 comment

Reply:
TODO: Move cloning before thread creation, lock manually in OVMS scenario